bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -374,12 +375,16 @@ crate::internal_macros::define_extension_trait! {
         #[deprecated(since = "TBD", note = "use `to_string()` instead")]
         fn to_asm_string(&self) -> String { self.to_string() }
 
-        /// Formats the script as lower-case hex.
+        /// Consensus encodes the script as lower-case hex.
+        fn to_hex_string(&self) -> String { consensus::encode::serialize_hex(self) }
```

Comment:
Strong NACK on doing it this way. This is a hard-to-debug behavioral breaking change - ironically the exact same debugging problem as you mention in the rationale.

I suggest instead adding `to_length_prefixed_hex_string` and deprecate `hex`.